Changelog.md

```diff
@@ -1,26 +1,39 @@
 # Change Log
 ## bpaf [0.6.0] - unreleased
-- removed `positional_os()` and `argument_os()`
-- removed `from_str`
-- internal refactors
-- documentation refactor: examples with input/output
-- `any` - a positional way to capture anything
-- `adjacent` restriction
-- `catch` inside `ParseOption`, `ParseMany` and `ParseSome`
-- usage line refactor, cosmetic mostly
-- include \n in --version, cosmetic
-
-Migration guide:
-1. replace `positional_os("FOO")` with `positional::<Type>("FOO")` or use `positional` in derive API
-2. replace `argument_os("FOO")` with `argument::<Type>("FOO")` or use `argument` in derive API
-3. replace any uses of `from_str::<T>` with turbofish on the nearest `positional` or
-   `argument`. You can use `map` to apply `FromStr` parser if `String` gets created inside your parsers
-
-You shouldn't be using those names directly, but there are some structure renames:
-1. `Positional` -> `ParsePositional`
-2. `BuildArgument` -> ParseArgument`
-3. `Command` -> `ParseCommand`
-
+# What's new in 0.6.0
+- `adjacent` restriction to parse things in a tighter context
+- `catch` for `many`, `some` and `optional` to handle parse errors
+- `any` positional like, to consume pretty much anything from a command line
+- improved documentation with more detailed examples
+- cosmetic improvements
+- a sneaky reminder to use `to_options` on `Parser` before trying to run it
+- a way to make boxed parsers with single item `construct!` macro for making dynamic parsers
+- a horrible way to reduce Yoda-talk coding by placing primitive definitions inside the `construct!` macro
+
+With new additions you should be able to parse pretty much anything and then some more :)
+
+# Migration guide 0.5.x -> 0.6.x
+
+1. Replace any uses of `positional_os` and `argument_os` with `positional` and `argument` plus turbofish:
+    ```diff
+    -let file = positional_os("FILE").help("File to use").map(PathBuf::from);
+    +let file = positional::<PathBuf>("FILE").help("File to use");
+    ```
+2. Replace any uses of `from_str` with either turbofish on the consumer or with `parse`, if `String` is generated inside the parser:
+    ```diff
+    -let number = short('n').argument("N").from_str::<usize>();
+    +let number = short('n').argument::<usize>("N");
+    ```
+    You can still use it for your own types if you implement `FromOsStr`, alternatively `parse` still works:
+    ```diff
+    -let my = long("my-type").argument("MAGIC").from_str::<MyType>();
+    +let my = long("my-type").argument::<String>("MAGIC").parse(|s| MyType::from_str(s));
+    ```
+3. You shouldn't be using those names directly in your code, but there are some renames
+    - `Positional` -> `ParsePositional`
+    - `BuildArgument` -> `ParseArgument`
+    - `Command` -> `ParseCommand`
+    - `Named` -> `NamedArg`
```

Comment:
Why `Parse` prefixes for the first three but not for `NamedArg`?

Personally, I would do
- `PositionalArg`
- `Argument`
- `Command`
- `NamedArg`

Reply:
> Why Parse prefixes for the first three but not for NamedArg?

First three are actually parsers that can extract something, `NamedArg` needs `.flag`, `.switch` or `.argument` to become one. At the moment any primitive parser starts with `Parse` so naming is consistent.

For `Command` - when messing with passing arguments to `cargo` I found it confusing when there's several things named `Command` sitting in one namespace: one from `bpaf` and one from `std::process::Command`. Having distinct names helps to reduce this kind of confusion. I guess I'll poke my coworkers as well for the input.

Reply:
I renamed `clap::App` to `clap::Command` back in [3.1.0](https://github.com/clap-rs/clap/blob/master/CHANGELOG.md#310---2022-02-16) and no one has brought up concerns over naming conflicts.  Granted, it was only deprecated and in clap 3.2 we made deprecations opt-in.  We'll see when clap v4 comes out.

Reply:
Well. For that to be reported you need a bunch of factors:
1. One needs to try to pass some stuff to a child process which might be a problem in general: https://github.com/clap-rs/clap/issues/1404
2. To do this in the same module as the argument parsing - I'm lazy and it's a prototype.
3. Be actually bothered enough to report. I probably wouldn't be.

Reply:
(1) seems irrelevant.  You just need to deal with `Command` in the same mod.  This can happen because of that, because of how `cargo run --` works, or just because

(2) depends on if you do `use clap::Command`.  Personally, I don't.  I only like using `use` for traits.